src/Core/Resolvers/DWSqlQueryBuilder.cs

```diff
@@ -0,0 +1,212 @@
+        }
+
+        /// <inheritdoc />
+        public string Build(SqlDeleteStructure structure)
+        {
+            throw new NotImplementedException("DataWarehouse sql currently does not support deletes");
+        }
+
+        /// <inheritdoc />
+        public string Build(SqlExecuteStructure structure)
+        {
+            throw new NotImplementedException("DataWarehouse sql currently does not support executes");
+        }
+
+        /// <inheritdoc />
+        public string Build(SqlUpsertQueryStructure structure)
+        {
+            throw new NotImplementedException("DataWarehouse sql currently does not support updates");
+        }
+
+        /// <summary>
+        /// Add a JSON_QUERY wrapper on the column
+        /// </summary>
+        private string WrapSubqueryColumn(LabelledColumn column, SqlQueryStructure subquery)
+        {
+            string builtColumn = Build(column as Column);
+            if (subquery.IsListQuery)
+            {
+                return $"JSON_QUERY (COALESCE({builtColumn}, '[]'))";
+            }
+
+            return $"JSON_QUERY ({builtColumn})";
+        }
+
+        /// <summary>
+        /// Build columns and wrap columns which represent join subqueries
+        /// </summary>
+        private string WrappedColumns(SqlQueryStructure structure)
+        {
+            return string.Join(", ",
+                structure.Columns.Select(
+                    c => structure.IsSubqueryColumn(c) ?
+                        WrapSubqueryColumn(c, structure.JoinQueries[c.TableAlias!]) + $" AS {QuoteIdentifier(c.Label)}" :
+                        Build(c)
+            ));
+        }
+
+        /// <inheritdoc />
+        public string BuildStoredProcedureResultDetailsQuery(string databaseObjectName)
+        {
+            throw new NotImplementedException("DataWarehouse sql currently does not support stored procedures");
+        }
+
+        /// <summary>
+        /// Builds the query to get all the read-only columns in an DWSql table.
+        /// For DWSql, the columns:
+        /// 1. That have data_type of 'timestamp', or
+        /// 2. are computed based on other columns,
+        /// are considered as read only columns. The query combines both the types of read-only columns and returns the list.
+        /// </summary>
+        /// <param name="schemaOrDatabaseParamName">Param name of the schema/database.</param>
+        /// <param name="tableParamName">Param name of the table.</param>
+        /// <returns>String representing the query needed to get a combined list of read only columns.</returns>
+        public string BuildQueryToGetReadOnlyColumns(string schemaParamName, string tableParamName)
+        {
+            // For 'timestamp' columns sc.is_computed = 0.
+            string query = "SELECT ifsc.column_name from sys.columns as sc INNER JOIN INFORMATION_SCHEMA.COLUMNS as ifsc " +
```

Comment:
Can this metadata query correctly identify read only columns? Do we have tests for DW?

Reply:
This is the same query used in mssql that works for DW. This query is tested as part of the integration tests for DW